lib/rules/default-param-last.js

```diff
@@ -5,16 +5,32 @@
 
 "use strict";
 
+/**
+ * Checks if node is required: i.e. does not have a default value or ? optional indicator.
+ * @param {ASTNode} node the node to be evaluated
+ * @returns {boolean} true if the node is required, false if not.
+ */
+function isRequiredParameter(node) {
+    return !(
+        node.type === "AssignmentPattern" ||
+        node.type === "RestElement" ||
+        node.optional
+    );
+}
+
 /** @type {import('../shared/types').Rule} */
 module.exports = {
     meta: {
+        dialects: ["javascript", "typescript"],
+        language: "javascript",
```

Comment:
Coming over from https://github.com/eslint/eslint/pull/19431#issuecomment-2672813876:

> * `meta.language = "javascript"` - to indicate that this rule is meant for JavaScript or a derivative of it
> * `meta.dialects = ["javascript", "typescript"]` - to indicate that this rule is known to work with both plain JavaScript and TypeScript.

Can I suggest a `meta.language: { dialects: string[]; family: string }` shape or something similar to that? If `dialects` is only ever meant to be used with `name`, I personally find it easier to reason about pushing them into an object together.

Reply:
The idea is that rules might only ever have `language` without `dialects` if they're narrowly focused.

Not opposed to revisiting elsewhere, I'd just like to stick with this to get started.

Reply:
https://github.com/eslint/eslint/issues/19462